app/grandchallenge/components/views.py

```
@@ -472,3 +494,114 @@ def get(self, request, *args, **kwargs):
             },
         )
         return HttpResponse(html_content)
+
+
+class FileWidgetSelectView(LoginRequiredMixin, View):
+    def get(self, request, *args, **kwargs):
+        interface = request.GET.get("interface-slug")
+        widget_choice = request.GET.get(f"widget-choice-{interface}")
+        help_text = request.GET.get("help-text")
+        current_value = request.GET.get("current-value")

```

Noticing here that you changed the names of the input fields from underscore to dash. Is there a specific reason for doing this? In general, I'm wondering whether the two widgets could make use of the same base templates to avoid duplication of the same logic. It would require adding some widget-specific context variables to the views, but should otherwise be possible, right?

---

On second thought, refactoring to use the same templates might break once you add the extra filter option for the FileWidget though

---

Thinking about the view design here, maybe it would be good to add some checks on whether the interface  exists and fail early if not: 

```
@cached_property
 def interface(self):
       return get_object_or_404(
            ComponentInterface, slug=self.request.GET.get("interface-slug") # why not use an underscore here?
       )
```

---

You could add a similar check for the widget choice 

```
@property
def widget_choice(self):
     try:
          return FileWidgetChoices(self.request.GET.get("widget-choice-(self.interface.slug)"))
     except ValueError:
         return Http404()
```

But you still need to check the widget_choice in `get()`, so not sure if this adds anything

---

And thinking about permission, currently, anyway can visit this view. I think it should be locked down to users who have access to (any) files on GC, similar to how we do it for `FileUploadFormFieldView`.

---

> you changed the names of the input fields from underscore to dash. Is there a specific reason for doing this?  

Not really, I initially started changing them to make sense of what was what and I found them dificult to read. I guess the mix of upper camelcase and lower camelcase combined with delimiter separation just triggered me :sweat_smile:  
I then tried to change everything to hyphen separated so I could more easily recognize when things were coming from the request data, and where it is used in the templates. 

> refactoring to use the same templates might break ...

Yes, I've just been updating the templates after some testing and there are some small differences already. After implementing the extra filter option we'll have to see. I'll continue with separate templates first and will take a look if I can reduce duplication when the work is done.

---

> maybe it would be good to add some checks on whether the interface exists and fail early if not

Thanks, I'll look into this!

---

Implemented checking for the interface existence and valid widget choice for failing early. I did not use a cached property because I only check it once in the get request. I also renamed data in the requests to `prefixed-interface-slug` for clarity.